plugins/slack/server/api/hooks.ts

```diff
@@ -278,90 +195,34 @@ router.post(
       return;
     }
 
-    // This should be super rare, how does someone end up being able to make a valid
-    // request from Slack that connects to no teams in Outline.
-    if (!team) {
+    const options = {
+      limit: 5,
+    };
+
+    if (!user) {
+      const team = await findTeamForRequest(team_id);
+
       ctx.body = {
         response_type: "ephemeral",
-        text: t(
-          `Sorry, we couldn’t find an integration for your team. Head to your {{ appName }} settings to set one up.`,
-          {
-            ...opts(user),
-            appName: env.APP_NAME,
-          }
-        ),
+        blocks: presentUserNotLinkedBlocks(team),
       };
       return;
     }
 
-    // Try to find the user by matching the email address if it is confirmed on
-    // Slack's side. It's always trusted on our side as it is only updatable
-    // through the authentication provider.
-    if (!user) {
-      const auth = await IntegrationAuthentication.findOne({
-        where: {
-          scopes: { [Op.contains]: ["identity.email"] },
-          service: IntegrationService.Slack,
-          teamId: team.id,
-        },
-      });
-
-      if (auth) {
-        try {
-          const response = await Slack.request("users.info", {
-            token: auth.token,
-            user: user_id,
-          });
-
-          if (response.user.is_email_confirmed && response.user.profile.email) {
-            user = await User.findOne({
-              where: {
-                email: response.user.profile.email,
-                teamId: team.id,
-              },
-            });
-          }
-        } catch (err) {
-          // Old connections do not have the correct permissions to access user info
-          // so errors here are expected.
-          Logger.info(
-            "utils",
-            "Failed requesting users.info from Slack, the Slack integration should be reconnected.",
-            {
-              teamId: auth.teamId,
-            }
-          );
-        }
-      }
-    }
-
-    const options = {
-      limit: 5,
-    };
-
-    // If we were able to map the request to a user then we can use their permissions
-    // to load more documents based on the collections they have access to. Otherwise
-    // just a generic search against team-visible documents is allowed.
-    const { results, totalCount } = user
-      ? await SearchHelper.searchForUser(user, text, options)
-      : await SearchHelper.searchForTeam(team, text, options);
+    const { results, totalCount } = await SearchHelper.searchForUser(
```

Comment:
No longer does `searchForTeam` if user is not known, instead ask them to connect account